### PR TITLE
Standardizing File, FTP/SFTP and HTTPS connector docs

### LINKED ISSUE #476: Enhance FTP/SFTP Data Connector documentation
- [x] Connector description
- [x] Configuration section
- [x] Secrets section / Authentication section
- [x] Examples section
- [x] Parameters table
- [x] File format support (if applicable)

### LINKED ISSUE #479: Enhance HTTP(s) Data Connector Documentation
- [x] Connector description
- [x] Configuration section
- [x] Secrets section / Authentication section
- [x] Examples section
- [x] Parameters table
- [x] File format support (if applicable)

### LINKED ISSUE #588: Enhance File connector documentation
Page should have:
- [ ] Connector description
- [ ] Configuration section
- [ ] Secrets section / Authentication section
- [ ] Examples section
- [ ] Parameters table
- [ ] File format support (if applicable)

### LINKED ISSUE #590: Enhance FTP/SFTP connector documentation
Page should have:
- [ ] Connector description
- [ ] Configuration section
- [ ] Secrets section / Authentication section
- [ ] Examples section
- [ ] Parameters table
- [ ] File format support (if applicable)

### PR DESCRIPTION
**Pull request by eadgbear**
- **Standardizing File connector documentation**
- **Enhancing HTTPS connector documentation**
- **Standardizing FTP connector documentation**

## 🗣 Description

Standardizing as outlined in #580 

## 🔨 Related Issues

- Closes #588 
- Closes #590
- Closes #479

### CODE REVIEW

File: spiceaidocs/docs/components/data-connectors/file.md [hunk 1]
```diff
@@ -52,3 +75,39 @@ datasets:
...
+
+```bash
+├── foo
+│   └── yellow_tripdata_2024-01.parquet
+└── spicepod.yaml
+```
+
+```yaml
+datasets:
+  - from: file:foo/yellow_tripdata_2024-01.parquet
```
Comment (sgrebnov) [inline spiceaidocs/docs/components/data-connectors/file.md:105]: Might be good to demo the following format as well (if supported)
```
- from: ./file:foo/yellow_tripdata_2024-01.parquet
```

**Review by sgrebnov — DISMISSED** [review]

**Comment by sgrebnov** [comment]
If we have good related quickstarts it might be good to link them as part of the docs IMO

**Comment by eadgbear** [comment]
> If we have good related quickstarts it might be good to link them as part of the docs IMO

That's a good point! I'll add it

File: spiceaidocs/docs/components/data-connectors/file.md [hunk 2]
```diff
@@ -4,8 +4,6 @@ sidebar_label: 'File Data Connector'
 description: 'File Data Connector Documentation'
 ---
 
-import Tabs from '@theme/Tabs';
-import TabItem from '@theme/TabItem';
 
 The File Data Connector enables federated SQL queries on files stored by locally accessible filesystems. It supports querying individual files or entire directories, where all child files within the directory will be loaded and queried.
```
Comment (phillipleblanc) [inline spiceaidocs/docs/components/data-connectors/file.md]:
```suggestion
The File Data Connector enables federated/accelerated SQL queries on files stored by locally accessible filesystems. It supports querying individual files or entire directories, where all child files within the directory will be loaded and queried.
```

File: spiceaidocs/docs/components/data-connectors/file.md [hunk 3]
```diff
@@ -52,3 +75,39 @@ datasets:
...
+
+```yaml
+datasets:
+  - from: file:foo/yellow_tripdata_2024-01.parquet
+    name: trip_data
+    params:
+      file_format: parquet
+```
+
+## Secrets
```
Comment (phillipleblanc) [inline spiceaidocs/docs/components/data-connectors/file.md]: Are secrets required for the file data connector?

File: spiceaidocs/docs/components/data-connectors/ftp.md [hunk 4]
```diff
@@ -4,69 +4,104 @@ sidebar_label: 'FTP/SFTP Data Connector'
 description: 'FTP/SFTP Data Connector Documentation'
 ---
 
-import Tabs from '@theme/Tabs';
-import TabItem from '@theme/TabItem';
+FTP (File Transfer Protocol) and SFTP (SSH File Transfer Protocol) are network protocols used for transferring files between a client and server, with FTP being less secure and SFTP providing encrypted file transfer over SSH.
 
-The FTP/SFTP Data Connector enables federated SQL query across Parquet/CSV files stored in FTP/SFTP servers.
+The FTP/SFTP Data Connector enables federated SQL query across [supported file formats](/components/data-connectors/index.md#object-store-file-formats) stored in FTP/SFTP servers.
```
Comment (phillipleblanc) [inline spiceaidocs/docs/components/data-connectors/ftp.md]:
```suggestion
The FTP/SFTP Data Connector enables federated/accelerated SQL query across [supported file formats](/components/data-connectors/index.md#object-store-file-formats) stored in FTP/SFTP servers.
```

File: spiceaidocs/docs/components/data-connectors/ftp.md [hunk 5]
```diff
@@ -4,69 +4,104 @@ sidebar_label: 'FTP/SFTP Data Connector'
...
 
-If a folder is provided, all child Parquet/CSV files will be loaded.
+```yaml
+datasets:
+  - from: sftp://remote-sftp-server.com/path/to/folder/
+    name: my_dataset
+    params:
+      file_format: csv
+      sftp_port: 20
```
Comment (phillipleblanc) [inline spiceaidocs/docs/components/data-connectors/ftp.md]:
```suggestion
      sftp_port: 22
```
Default sftp port is 22

File: spiceaidocs/docs/components/data-connectors/ftp.md [hunk 6]
```diff
@@ -4,69 +4,104 @@ sidebar_label: 'FTP/SFTP Data Connector'
...
-          sftp_user: my-sftp-user
-          sftp_pass: ${secrets:my_sftp_password}
-          hive_partitioning_enabled: true
-    ```
-
-  </TabItem>
-</Tabs>
+### `from`
+
+The `from` field takes one of two forms: `ftp://path` or `sftp://path` where `path` is the path to the file or directory to read from.
```
Comment (phillipleblanc) [inline spiceaidocs/docs/components/data-connectors/ftp.md]: Its two parts, the `host` and the `path` within that host, which is in contract to something like databricks where the host is not part of the `from` field.

File: spiceaidocs/docs/components/data-connectors/https.md [hunk 7]
```diff
@@ -5,26 +5,76 @@ description: 'HTTP(s) Data Connector Documentation'
 pagination_prev: null
 ---
 
-The HTTP(s) Data Connector enables federated SQL query against a variety of tabular formatted (e.g. Parquet/CSV) files stored at a HTTP endpoint.
+The HTTP(s) Data Connector enables federated SQL query across [supported file formats](/components/data-connectors/index.md#object-store-file-formats) stored at an HTTP(s) endpoint.
```
Comment (phillipleblanc) [inline spiceaidocs/docs/components/data-connectors/https.md]:
```suggestion
The HTTP(s) Data Connector enables federated/accelerated SQL query across [supported file formats](/components/data-connectors/index.md#object-store-file-formats) stored at an HTTP(s) endpoint.
```

File: spiceaidocs/docs/components/data-connectors/file.md (hunk 1 above)
Comment (eadgbear) [reply spiceaidocs/docs/components/data-connectors/file.md:105]: I don't think that method is supported

File: spiceaidocs/docs/components/data-connectors/file.md (hunk 3 above)
Comment (eadgbear) [reply spiceaidocs/docs/components/data-connectors/file.md]: Removed, good point.

File: spiceaidocs/docs/components/data-connectors/ftp.md (hunk 5 above)
Comment (eadgbear) [reply spiceaidocs/docs/components/data-connectors/ftp.md]: Fixed, thank you

File: spiceaidocs/docs/components/data-connectors/ftp.md (hunk 6 above)
Comment (eadgbear) [reply spiceaidocs/docs/components/data-connectors/ftp.md]: Fixed

**Review by sgrebnov — APPROVED** [review]

**Pull request merged**